Add count_include_pad arg for PoolOpGradient on CPU and fix ARM performance issue.

xiaomengy opened the pull request:
Summary: Add count_include_pad arg for PoolOpGradient on CPU and fix ARM performance issue.

Differential Revision: D13564257

houseroad commented on caffe2/operators/pool_op.cc:
```diff
@@ -1,35 +1,603 @@
+        ? EigenArrayMap<T>(Y_ptr, Y_HxW, C)
+        : EigenArrayMap<T>(Y_ptr, C, Y_HxW);
+    for (int d = 0; d < Y_D; ++d) {
+      const int p = std::max(d * stride_d - pad_p, 0);
```
Nit: maybe we can have some inline function or macro to represent the following two common patterns:
- max(x * stride - pad, 0)
- min(x * stride - pad + kernel, X)

houseroad reviewed (approved):
Looks good, please add the tests for `count_include_pad` in the following PRs, thanks!

BIT-silence commented:
@pytorchbot retest this please

xiaomengy committed:
Add count_include_pad arg for PoolOpGradient on CPU and fix ARM performance issue. (#15651)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/15651

Add count_include_pad arg for PoolOpGradient on CPU and fix ARM performance issue.

Reviewed By: houseroad

Differential Revision: D13564257

fbshipit-source-id: c6f3b264969b91d61211142cfc76087560731df5
